proposals/lip-0055.md

```diff
@@ -105,16 +128,44 @@ The full processing of a block is organized as follows.
 5. **Assets verification**: Each module verifies the respective entry in the block assets. If any check fails, the block is discarded and has no further effect.
 6. **Block forwarding**: After the initial checks, the full block is forwarded to a subset of peers.
 7. **Before transactions execution**: Each module can define protocol logic that is executed *before* the transactions contained in the block are processed.
-8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If the verification fails, the transaction is invalid and the whole block is discarded.
-9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified.
-10. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed.
-11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed.
-12. **After command execution**: Each module can define protocol logic that is processed *after* the command has been executed.
+8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If an error occurs, the transaction is invalid and the whole block is discarded.
+9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified. If an error occurs, the transaction is invalid and the whole block is discarded.
+10. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed. If an error occurs, the transaction is invalid and the whole block is discarded.
+11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed. If an error occurs, the transaction is failed and the processing continues. All state transitions performed in this stage are reverted.
```

Comment:
```suggestion
11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed. If an error occurs, the transaction is failed and all state transitions performed in this stage are reverted. In any case, afterwards the processing continues with the next stage.
```

Reply:
To make it more explicit that the transitions are only reverted in case of an error.